src/main/studio/ui/SearchPanel.java

```diff
@@ -73,11 +74,17 @@ public SearchPanel(EditorPane editorPane) {
 
         ActionMap am = txtFind.getActionMap();
         InputMap im = txtFind.getInputMap();
+        int shift = InputEvent.SHIFT_DOWN_MASK;
         am.put("findAction", findAction);
         am.put("closeAction", closeAction);
         im.put(KeyStroke.getKeyStroke(KeyEvent.VK_ENTER,0),"findAction");
         im.put(KeyStroke.getKeyStroke(KeyEvent.VK_ESCAPE,0),"closeAction");
 
+        am.put("findNextAction", findAction);
+        am.put("findPreviousAction", findBackAction);
+        im.put(KeyStroke.getKeyStroke(KeyEvent.VK_F3,0),"findNextAction");
```

Comment:
This one is ok. However I think that you can map several keystrokes to the same action... So you can remove line 83 and replace this one with 
```
im.put(KeyStroke.getKeyStroke(KeyEvent.VK_F3,0),"findAction");
```